hipscat/util.py

```diff
@@ -289,148 +286,111 @@ def xmatchmap_dict(hp_match_map):
     return data
 
 
-def catalog_columns_selector_withdtype(cat_md, cols):
+def rename_meta_cols(c1_cols, c2_cols, suffix='_2'):
+    '''
+        finds columns with the same name and appends
+        a suffix to the column name. Should only be applied
+        to the c2_cols dictionary
     '''
-        Establish the return columns for the dataframe's metadata
-         dask.dataframe.map_partitions() requires the metadata of the resulting 
-         dataframe to be defined prior to execution. The column names and datatypes
-         are defined here and passed in the 'meta' variable
+    for c2c in c2_cols:
+        if c2c in c1_cols:
+            c2c_i = c2_cols.index(c2_cols[c2c])
+            c2_cols[c2c_i] = f'{c2c}{suffix}'
+    return c2_cols
 
-         it expects the ra_kw, dec_kw, and id_kw fields with their respective dtypes
-         if the user want's other columns, it will append them
 
-         TODO: validate user-defined cols fields {key: dtype} 
+def cater_input_cols(cols, cat_md):
     '''
-    expected_cols = {
-        cat_md['ra_kw'] :'f8',
-        cat_md['dec_kw']:'f8',
-        cat_md['id_kw'] :'i8'
-    }
-    if not len(cols):
-        return expected_cols
-    else:
-        for k in expected_cols.keys():
-            if k not in cols.keys():
-                cols[k] = expected_cols[k]
+    if the user specifies columns in a crossmatch 
+        this ensures they don't forget the ra,dec,and id
+    if they don't specify columns, the read_parquet method
+        will read all columns
+    '''
+
+    if len(cols):
+        expected_cols = [
+            cat_md['ra_kw'],
+            cat_md['dec_kw'],
+            cat_md['id_kw'],
+        ]
+        
+        for ec in expected_cols:
+            if ec not in cols:
+                cols.append(ec)
+
         return cols
+    return None
 
 
-def establish_pd_meta(c1_cols, c2_cols):
+def frame_prefix_all_cols(df, prefix):
     '''
-        the return type from cross_match routine metadata columns
+    appends a prefix to all columns in a dataframe
     '''
-    colnames = []
-    colnames.extend(c1_cols)
-    colnames.extend(c2_cols)
-    colnames.extend(['hips_k', 'hips_pix', '_DIST'])
-    return colnames
+
+    cols = list(df.columns)
+    rename_dict = {}
+    for d in cols:
+        rename_dict[d] = f'{prefix}_{d}'
```

Comment:
I thought we'd discussed and decided on dot notation, to align with SQL-like column name prefixing.

Reply:
This has been implemented now. prefixes are now `catname.column`.